lib/posix-fdio/fdstat.c

```diff
@@ -0,0 +1,122 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+/* Internal syscalls for manipulating file metadata */
+
+#include <uk/posix-fdio.h>
+
+#include "fdio-impl.h"
+
+static inline
+dev_t nums2dev(__u32 major, __u32 minor)
+{
+	return ((dev_t)major << 32) | minor;
+}
+
+static inline
+void timecpy(struct timespec *d, const struct uk_statx_timestamp *s)
+{
+	d->tv_sec = s->tv_sec;
+	d->tv_nsec = s->tv_nsec;
+}
+
+static inline
+void statx_cpyout(struct stat *s, const struct uk_statx *sx)
+{
+	unsigned int mask = sx->stx_mask;
+
+	s->st_dev = nums2dev(sx->stx_dev_major, sx->stx_dev_minor);
+	s->st_rdev = nums2dev(sx->stx_rdev_major, sx->stx_rdev_minor);
+	s->st_blksize = sx->stx_blksize;
+	if (mask & STATX_INO)
```

Comment:
Ouh... we need to update all of these `STATX_*` fields, I will do this while upstreaming.